webapp/platform/client/src/client4.ts

```diff
@@ -2062,7 +2062,7 @@ export default class Client4 {
 
     // Post Routes
 
-    createPost = async (post: Post) => {
+    createPost = async (post: Partial<Post> & {channel_id: string}) => {
```

Comment:
We have a `PartialExcept` type in `@mattermost/types/utilities` which I think would be useful here. Would you mind changing that?